enhancements/monitoring/alert-overrides.md

```diff
@@ -0,0 +1,576 @@
+	// +optional
+	ValidationErrors []string `json:"validationErrors,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// AlertRelabelConfigList is a list of AlertRelabelConfigs.
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+// +k8s:openapi-gen=true
+type AlertRelabelConfigList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	// items is a list of AlertRelabelConfigs.
+	Items []*AlertRelabelConfig `json:"items"`
+}
+
+// LabelName is a valid Prometheus label name which may only contain ASCII
+// letters, numbers, and underscores.
+//
+// +kubebuilder:validation:Pattern:="^[a-zA-Z_][a-zA-Z0-9_]*$"
+type LabelName string
+
+// RelabelConfig allows dynamic rewriting of label sets for alerts.
+// See Prometheus documentation:
+// - https://prometheus.io/docs/prometheus/latest/configuration/configuration/#alert_relabel_configs
+// - https://prometheus.io/docs/prometheus/latest/configuration/configuration/#relabel_config
+//
+// +k8s:openapi-gen=true
+type RelabelConfig struct {
+	// sourceLabels select values from existing labels. Their content is
+	// concatenated using the configured separator and matched against the
+	// configured regular expression for the replace, keep, and drop actions.
+	//
+	// +optional
+	SourceLabels []LabelName `json:"sourceLabels,omitempty"`
+
+	// separator placed between concatenated source label values. When omitted,
+	// Prometheus will use its default value of ';'.
+	//
+	// +optional
+	Separator string `json:"separator,omitempty"`
+
+	// targetLabel to which the resulting value is written in a replace action.
+	// It is mandatory for 'replace' and 'hashmod' actions. Regex capture groups
+	// are available.
+	//
+	// +optional
+	TargetLabel string `json:"targetLabel,omitempty"`
+
+	// regex against which the extracted value is matched. Default is: '(.*)'
+	//
+	// +optional
+	Regex string `json:"regex,omitempty"`
+
+	// modulus to take of the hash of the source label values.  This can be
+	// combined with the 'hashmod' action to set 'target_label' to the 'modulus'
+	// of a hash of the concatenated 'source_labels'.
+	//
+	// +optional
+	Modulus uint64 `json:"modulus,omitempty"`
+
+	// replacement value against which a regex replace is performed if the regular
+	// expression matches. This is required if the action is 'replace' or
+	// 'labelmap'. Regex capture groups are available. Default is: '$1'
+	//
+	// +optional
+	Replacement string `json:"replacement,omitempty"`
+
+	// action to perform based on regex matching. Must be one of: replace, keep,
+	// drop, hashmod, labelmap, labeldrop, or labelkeep.  Default is: 'replace'
+	//
+	// +kubebuilder:validation:Enum=replace;keep;drop;hashmod;labelmap;labeldrop;labelkeep
+	// +kubebuilder:default=replace
```

Comment:
While I appreciate this deviates from what the upstream API does, Kube conventions state that these should be Pascal case, so these should be 
```suggestion
	// +kubebuilder:validation:Enum=Replace;Keep;Drop;HashMod;LabelMap;LabelDrop;LabelKeep
	// +kubebuilder:default=Replace
```
It may seem odd but we are trying to keep the pattern consistent for OpenShift users, they expect any enumeration or option to be Pascal case

Reply:
Done. I honestly think this is worse UX if these are case sensitive. I think the best UX would be if a user can copy a `PrometheusRule` that doesn't contain recording rules as-is and have it work as an `AlertingRule`. But I just want to move this forward one way or another.